### ISSUE TITLE
Controllable filterFields and filters

### ISSUE THREAD

**Issue by GoGoCarl** [digidem/react-mapfilter#67]
Feature Request:

Currently, it seems that the `FilterPane` can only be populated using a best guess algorithm or by explicit user selection via settings. Allow the MapFilter props for `filterFields` and `filters` to be controllable. The idea behind this would be to:

* Supply the MapFilter component with a set of filters that initially appear in the `FilterPane`
* This set of initial filters should be able to take an initial value as well, i.e. checked or not checked for discrete filters, or a date range (value and min/max) for date filters.
* Consumers should be able to listen for changes to these filters.

**Comment by GoGoCarl** [comment]
@karissa Setting the filters works fine, and I can set the filter fields initially, but I can no longer modify the filters in the "change filters" interface. Also, the file `util/stateReconciler.js` references `controllableProps`, but it is not defined in that file.

**Comment by okdistribute** [comment]
@GoGoCarl did you listen to the respective onChange events and call setState with the new values? This will reflect the user's update filter actions back to the UI

**Comment by okdistribute** [comment]
For example,

```

  handleChangeFilterFields = (filterFields) => {
    this.setState({filterFields})
  }

  handleChangeFilters = (filters) => {
    this.setState({filters})
  }
```

**Comment by okdistribute** [comment]
@GoGoCarl oops, refactored stateReconciler out and missed a parameter. Fixed on latest next branch de678f8

**Comment by GoGoCarl** [comment]
@karissa Yes, I did precisely that. I'll give it a try with the latest branch and see if that helps.
